src/core_modules/capture-core-utils/rulesEngine/services/VariableService/helpers/structureEvents.js

```diff
@@ -0,0 +1,44 @@
+// @flow
```

Comment:
Added some missing parts to the rules engine variable service. Also, some common logic that earlier resided outside of the rules engine, I moved into he engine itself.